sync2/types/types.go

```diff
@@ -0,0 +1,263 @@
+package types
```

Comment:
We can't have these types in `sync2` package as it will be acting as a facade for `sync2/rangesync`, `sync2/multipeer` and `sync2/dbsync` packages, and we can't have cyclic dependencies.
My initial idea was for `sync2/multipeer` package was not to depend directly on `sync2/rangesync` package, but it turned out not to be doable unless I move the whole `OrderedSet` thing into `sync2/types`, too.
So I should probably just move these types back to `sync2/rangesync` package

Reply:
Merged `sync2/types` into `sync2/rangesync`, in the end it seems to be an easier way to go to avoid having extremely tiny packages